### PR TITLE
brial: init at 1.2.3

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

~~This depends on #38762, do not merge before that~~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/science/math/brial/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  nativeBuildInputs = [
+    autoreconfHook
+    pkgconfig
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/BRiAl/BRiAl;
+    description = "Legacy version of PolyBoRi maintained by sagemath developers";
+    longDescription = ''
+      M4RIE is a library for fast arithmetic with dense matrices over small finite fields of even characteristic. It uses the M4RI library, implementing the same operations over the finite field F2.
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/brial/default.nix]: wrong `longDescription`

**Comment by timokau** [comment]
I just tried to nox-review this and it failed, don't merge.

**Comment by timokau** [comment]
Resolved. I had to disable the boost test framework, since thats not packaged.

File: pkgs/development/libraries/science/math/brial/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "--with-boost-unit-test-framework=no" # boost-test is not packaged
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkgconfig
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/BRiAl/BRiAl;
```
[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/development/libraries/science/math/brial/default.nix]: I think that is less clear than explicitly stating it. Its just more or less a coincidence that those two are the same.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build brial

**Comment by dotlambda** [comment]
There are tests in https://github.com/BRiAl/BRiAl/tree/master/tests. I'm not sure whether it makes sense to run these.

**Comment by dotlambda** [comment]
Since this is the "legacy version of PolyBoRi maintained by sagemath developers", it should be put in the overlay.

**Comment by timokau** [comment]
The tagline is wrong / confusing: brial is the new one, polybori is legacy. The github short description is "Successor to PolyBoRi".

File: pkgs/development/libraries/science/math/brial/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  buildInputs = [
+    boost
+    m4ri
+    gd
+  ];
+
+  configureFlags = [
+    "--with-boost-unit-test-framework=no" # boost-test is not packaged
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/brial/default.nix]: Testing would probably reduce the necessary maintenance effort. So if it isn't too difficult, I'd suggest packaging `boost-test` and adding `doCheck = true`.

Comment (timokau) [reply pkgs/development/libraries/science/math/brial/default.nix]: I've done that (`boost init` commit is now included in this PR).

Comment (dotlambda) [reply pkgs/development/libraries/science/math/brial/default.nix]: Thanks a lot!

File: pkgs/development/libraries/boost-test/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  # https://www.boost.org/doc/libs/1_67_0/more/getting_started/unix-variants.html
+  installPhase = ''
+    mkdir -p $out
+    cp -r include $out/include
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://boost.org/libs/test;
+    description = "Boost.org test module";
+    license = {
```
Comment (dotlambda) [inline pkgs/development/libraries/boost-test/default.nix]: There is `licenses.boost`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by timokau** [comment]
Again, not ready for merge (I only tested the build non-sandboxed with boost-test before pushing, doesn't build when sandboxed).

**Comment by timokau** [comment]
I'm having issues with `boost-test`. First I thought just copying would be enough (since the boost docs indicate sometiimes it is and `boost-process` is packaged that way). Unfortunately its not.

The build fails here:

```
Scanning dependencies of target doc-example53
[ 72%] Building CXX object CMakeFiles/doc-example53.dir/tmp/nix-build-boost-test-1.66.0.drv-0/source/doc/examples/example53.run-fail.cpp.o
[ 72%] Linking CXX executable static-library-custom-init-test
/nix/store/27x7pinqdsl9f3rpbm8bsszd9fhwq266-glibc-2.27/lib/crt1.o: In function `_start':
/build/glibc-2.27/csu/../sysdeps/x86_64/start.S:104: undefined reference to `main'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/static-library-custom-init-test.dir/build.make:96: static-library-custom-init-test] Error 1
make[1]: *** [CMakeFiles/Makefile2:5806: CMakeFiles/static-library-custom-init-test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 72%] Linking CXX executable doc-example52
[ 72%] Built target doc-example52
[ 73%] Linking CXX executable doc-example51
[ 73%] Built target doc-example51
[ 74%] Linking CXX executable doc-example53
[ 74%] Built target doc-example53
make: *** [Makefile:95: all] Error 2
builder for '/nix/store/r9x5zbg7g01zlb1yzgqxksac7f9x83pg-boost-test-1.66.0.drv' failed with exit code 2
```

I don't know much about cmake or boost. Do you have an idea what the error might be? I pushed the update to `boost-test` (still not mergeable).

**Comment by timokau** [comment]
Since I don't know how to proceed and apparently nobody else does either I reverted the `boost-test` change.

File: pkgs/development/libraries/science/math/brial/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  src = fetchFromGitHub {
+    owner = "BRiAl";
+    repo = "BRiAl";
+    rev = version;
+    sha256 = "0qy4cwy7qrk4zg151cmws5cglaa866z461cnj9wdnalabs7v7qbg";
+  };
+
+  configureFlags = [
+    "--with-boost-unit-test-framework=no"
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/brial/default.nix]: You might want to add
```
# FIXME: package boost-test and enable doCheck
```
here and then it's good to go.

**Comment by timokau** [comment]
Done

**Pull request merged**